docs/release-process.md

```diff
@@ -4,17 +4,39 @@
 We use [cargo-release](https://crates.io/crates/cargo-release) to simplify the release process.
 (We rely on v0.22 or later because it has support for workspaces. Install this with
 `cargo install cargo-release`, not to be confused with the different `cargo install release`!)
-It's not (yet) quite an ideal fit for our workflow, but it helps! Steps:
+It's not (yet) quite an ideal fit for our workflow, but it helps!
 
-1. Take a look over `CHANGELOG.md` and make sure the "unreleased" section accurately reflects the
-   contents of the new release.
-   * Anything that could cause a consumer project to behave differently if it upgraded
-     to the new version, should be called out as a breaking change and should trigger
-     a minor version bump (we're below `v1.0` for semver purposes).
+We use a separate version number for `uniffi` compared to all the other crates:
+
+  - The `uniffi_*` crates are intended for either internal use or by external bindings generators.
+    These crates get breaking version bumps regularly.
+  - `uniffi` acts as a frontend to the other crates and re-exports the top-level UniFFI
+    functionality.  `uniffi` gets breaking version bumps less often than the other crates (see below
+    for details).
+
+Steps:
+
+1. Take a look over `CHANGELOG.md` and make sure the "unreleased" section lists all changes for
+   the release.
+   * Anything that affects any UniFFI consumers should be listed, this includes consumers that
+     use UniFFI to generate their scaffolding/bindings, external bindings generators, etc.
+
+1. Take a look over `uniffi/CHANGELOG.md` and make sure the "unreleased" section lists changes for
+   the release that affect users `uniffi` of the UniFFI crate.
+   * This should be a copy of the items from the top-level `CHANGELOG.md` that affects
+     scaffolding/bindings generation.
+   * Note that breaking changes for a particular `uniffi_*` crate are not necessarily breaking for
+     the `uniffi` crate.  See `./uniffi-versioning.md` for a discussion of this.
```

Comment:
Having to maintain two separate changelogs doesn't seem great, but I can't think of a better way to handle this.

AFAICT, the changelogs are not published anywhere they're only available by browsing github.  If this is true, it's nice because I don't need to update any automation.  But maybe we should consider including the changelog in the docs.

Reply:
I don't understand why we need a new CHANGELOG, and don't understand how I'd actually use both (ie, *every* change to the sub-crates potentially impacts users of the top-level crate, as many features will be released without a top-level version bump. I assume this has something to do with communicating how breaking a breaking change is?

Reply:
Some changes will only affect the sub crates.  Here's some examples from the current CHANGELOG:
 -  The `guess_crate_root` function is now public
- `uniffi_bindgen`: Renamed `FFIArgument`, `FFIFunction` and `FFIType` to `FfiArgument`, `FfiFunction` and `FfiType` (this was a breaking change).

The issue you mentioned was also on my mind: a change could be breaking for sub-crate users, but non-breaking for top-level users.  However, when I looked through the changelog, I didn't see any examples of those.

The last issue was version numbers.  The version numbers for `uniffi` and the sub-crates are going to start to diverge, which makes the changelog tricky.  I don't know how we would setup the `cargo release` template to move changes from the unreleased section, if there are 2 different version numbers involved.